### ISSUE TITLE
[$250] Menu - The font size (menu text) does not respond to resizing

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46902]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.17-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** sustinov@applausemail.com
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/46707
## Action Performed:
Change the font size in OS to the maximum font size. In Chrome settings (Accessibility), increase text scaling to maximum
1. Open in Chrome https://staging.new.expensify.com/
2. Log in with any login
3. Navigate to the settings
4. Navigate to About
5. Navigate through the different menus
6. Watch the font size

## Expected Result:
~Text size in mWeb(Chrome) app partially does not resize~
The text size in mWeb(Chrome) app should react to the increase in size.

## Actual Result:
The font size (menu text) does not respond to resizing

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/900d26a9-e121-4b0a-bd91-1757617c0816

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c57c6c3405a2d684</li>
        <li>Upwork Job ID: 1821663436717148240</li>
        <li>Last Price Increase: 2024-09-19</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@johncschuster  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
he font size (menu text) does not respond to resizing

### What is the root cause of that problem?
We need to set the font size via `getValueUsingPixelRatio` and in all places where we gave static font size it will not rescale
### What changes do you think we should make in order to solve the problem?

We need to set the font size via `getValueUsingPixelRatio` every where we need
### What alternative solutions did you explore? (Optional)

**Comment by muhammad-r39** [comment]
Hi, the issue seems to be occurring due to using fixed pixel value and that is why its not responding to device settings. Using rem (root em) value instead of pixel will fix this issue. Please see the attached screenshot. I manually changed the value in developer mode to a single menu item. The solution is to update the script thats generating fontsize and update there.

![Screenshot_1](https://github.com/user-attachments/assets/a9ed8aed-2bf4-4ef5-a5bf-802e0054c05c)

**Comment by muhammad-r39** [comment]
Contributor details
Your Expensify account email: muhammadrussell39@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/muhammadrussell

**Comment by muhammad-r39** [comment]
Here is the screen recording. 
(Note: its manually added to one item only)

https://github.com/user-attachments/assets/f18d52cc-ac67-4512-b6ee-4dff92556fbc

**Comment by situchan** [comment]
Awaiting proposals

**Comment by saowwapark** [comment]
Contributor details
Your Expensify account email: saowwapark.chanwigoon@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~012348c9a4780fd260?mp_source=share

By default, the font-size in React is `pixel` unit which is static unit. To change the size depending on accessibility control please change it to be `rem` unit. It is a convention to use `rem` unit for font size in modern web application.

The real problem is to change from `pixel` to `rem` throughout the project is rather cumbersome. You have to decide whether which method you prefer between ...

1) Setting root font size (Not recommended)
```
html {
	font-size: 62.5%;
}

.paragraph-font-size {
    /* The font size for default is 18px. */
    font-size: 1.8rem;
}
```
If you want the size 18px, you have to set it to be `1.8rem`. 

2) CSS variable (Recommended)
```
html {
    --18px: 0.875rem
}

.paragraph-font-size {
    /* The font size for default is 18px. */
    font-size: var(--18px);
}
```

**Comment by situchan** [comment]
@saowwapark Can you please familiarize yourself with our codebase and some old proposals on closed issues to understand what is expected in a proposal?

**Comment by situchan** [comment]
> We need to set the font size via getValueUsingPixelRatio every where we need

@FitseTLT thanks for the proposal. can you please list all places? The test branch is fine.

**Comment by FitseTLT** [comment]
> > We need to set the font size via getValueUsingPixelRatio every where we need
> 
> @FitseTLT thanks for the proposal. can you please list all places? The test branch is fine.

@situchan It looks like the problem is specific to mWeb and don't think my RCA is correct.

**Comment by johncschuster** [comment]
Bump

**Comment by muhammad-r39** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Menu - The font size (menu text) does not respond to resizing
### What is the root cause of that problem?
The root cause lies in how Chrome and Safari handles browser/device font settings. In Safari/ios it works because safari respects users preference settings and adjust/scales the font size accordingly. Where for chrome (whether its ios or android) it requires some specific CSS to do that i.e setting the font-size to rem/em value. And in this case the font size is getting rendered in fixed px value in `Text.tsx` Component from the `componentStyle` object. So Android/Chrome will not adjust to the browser settings as its a fixed px value.
### What changes do you think we should make in order to solve the problem?
In the `Text.tsx` Component first we need to add a simple function that would convert the px value to rem. Then we need to check if the platform is web using the `Platform` component from `react-native` and conditionally convert the font-size to rem only for web. And it should work fine. I have made the changes and tested it in my local setup. Please see the attached video.
 

https://github.com/user-attachments/assets/4f5e5576-932f-4298-a941-db60216ed698

**Comment by muhammad-r39** [comment]
## Proposal 
[Updated] https://github.com/Expensify/App/issues/46902#issuecomment-2294531715

**Comment by muhammad-r39** [comment]
[Updated] https://github.com/Expensify/App/issues/46902#issuecomment-2294531715

Can I please get some feedback on it.

**Comment by situchan** [comment]
@muhammad-r39 thanks for the proposal. Have you tested your fix on all platforms?

**Comment by muhammad-r39** [comment]
@situchan previously I only tested for web, but now I have tested and added screen recordings of my test results for Android and iOS as well. Please see the new videos I have added in the original proposal.

[Updated] https://github.com/Expensify/App/issues/46902#issuecomment-2294531715

**Comment by FitseTLT** [comment]
The problem is on Android mWeb.

**Comment by muhammad-r39** [comment]
@FitseTLT since @situchan asked for the test on all platforms and when I tested Android and iOS app I noticed that the issue is also happening there. If the change is not needed for Android and iOS then the issue I talked about "font scaling is disabled" can be discarded. And we can leave it as it is.

And to apply the changes to mWeb only my initial suggestion should cover that. We can do a Platform check and if it's 'web' then use the function that would convert the px into rem value.

**Comment by muhammad-r39** [comment]
@johncschuster @situchan did you see my update?

Here is the short summary one more time: I tested and it works on both Safari and Chrome in web and mWeb. 
My solution is: Using a function that would convert the px into rem and conditionally apply it only for web (mWeb).

Additionally when I tested Android and iOS app I noticed that the issue is also happening there. The reason is font scaling option is disabled. If its working on current production app then we can discard this part.

Please let me know if you have any questions.

**Comment by johncschuster** [comment]
@situchan^^

**Comment by situchan** [comment]
checking...

**Comment by situchan** [comment]
android chrome:
<img width="197" alt="image" src="https://github.com/user-attachments/assets/ac960ab0-35e1-41ae-8375-ea5020c9d13a">

android native:
<img width="195" alt="image" src="https://github.com/user-attachments/assets/c559c1ca-69c4-4dde-bd93-8c43734763d7">

I am a bit confused about the expected behavior.
So mChrome should be same as android native (all app font sizes are increased responding to system font size) to be consistent?

cc: @johncschuster

**Comment by muhammad-r39** [comment]
@situchan the issue is how each platform handles font-size. The reason it's not working in Chrome is because we are passing a fixed px value font-size. And when we do that, Chrome gives priority to the fixed size and will not respect the device preference/ accessibility settings. The work around is using rem value instead of px for font-size. Safari will support both px and rem.

But for Android Native and iOS we need to use px.

This is why in my solution I mentioned a Platform check and apply the conversion of px to rem function only in web.

Please let me know if the explanation helps.

**Comment by johncschuster** [comment]
@Expensify/design I'd love your thoughts on [this comment](https://github.com/Expensify/App/issues/46902#issuecomment-2306326314). I would tend to believe that we would want all platforms to respond similarly when the system font is increased/decreased. Am I right to assume that? Are there any upper/lower limits to that I'm not considering?

**Comment by dubielzyk-expensify** [comment]
I think all font sizes on that screen should react to the increase in size. It feels weird to me as well that the headline isn't changing.

**Comment by shawnborton** [comment]
Yup, I agree with that.

**Comment by dannymcclain** [comment]
Same - I agree with Jon & Shawn.

**Comment by situchan** [comment]
Please update proposals based on https://github.com/Expensify/App/issues/46902#issuecomment-2311324437.
Make sure to test your solutions on all platforms.
Thanks

**Comment by muhammad-r39** [comment]
[Updated] https://github.com/Expensify/App/issues/46902#issuecomment-2294531715
@situchan

**Comment by muhammad-r39** [comment]
[Updated] https://github.com/Expensify/App/issues/46902#issuecomment-2294531715

**Comment by muhammad-r39** [comment]
[Updated] https://github.com/Expensify/App/issues/46902#issuecomment-2294531715

**Comment by situchan** [comment]
> Enable font scaling in Text.tsx file on line 50.
> allowFontScaling={false}

@muhammad-r39 This will cause so many regressions. We intentionally added this line in the past. Please find out PR which added this and regressions accordingly.

**Comment by muhammad-r39** [comment]
@situchan thanks for the information. I just have one question to make sure we are on the same page. The change need to happen not only mWeb but all Platforms (including Android and iOS), right?

**Comment by situchan** [comment]
> @situchan thanks for the information. Can you please confirm one more time that, the change need to happen not only mWeb but all Platforms (including Android and iOS).

correct since android/iOS still don't have the issues

**Comment by dubielzyk-expensify** [comment]
I'm not 100% sure actually. I would check the rest of that screen and stop the font-size when the rest is stopping. 

I think for a headline on that screen specifically I'd stop at around 40px. 

I'm making a note for the future to start defining this stuff 👍  Thanks for calling it out and asking 😄

**Comment by johncschuster** [comment]
I think we're waiting for a bit of clarification on this one, Melv.

**Comment by situchan** [comment]
Not overdue

**Comment by johncschuster** [comment]
Bumping this to keep Melvin happy. It looks like we're waiting on a proposal to be accepted.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by johncschuster** [comment]
Thanks for the retest!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by situchan** [comment]
Not overdue

**Comment by situchan** [comment]
Not overdue

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by situchan** [comment]
OP needs updated to meet https://github.com/Expensify/App/issues/46902#issuecomment-2311324437 on all platforms

**Comment by johncschuster** [comment]
Updated

![2024-09-25_14-13-06](https://github.com/user-attachments/assets/58530bf0-305b-47cb-a46a-a46d6c588c3d)
